tests/e2e/steps/dicomViewerContext.ts

```diff
@@ -1,47 +1,51 @@
-import {Given, When, Then} from '@cucumber/cucumber'
-import {state} from '../hooks'
-import {config} from "../config.js"
+import { Given, When, Then } from '@cucumber/cucumber'
+import { state } from '../hooks'
+import { config } from "../config.js"
 
-import {DicomViewer} from "../pageObjects/DicomViewer"
-import util from "util";
+import { DicomViewer } from "../pageObjects/DicomViewer"
+import { getUser } from "../userStore";
 
-Given('{string} has logged in', async function(user: string): Promise<void> {
+Given('the user {string} has logged in', async function(user: string): Promise<void> {
     const page = state.page
     const dicomViewer = new DicomViewer()
-
     await page.goto(config.baseUrlOcis)
-    await dicomViewer.login({ username: 'admin', password: 'admin' })
-    await page.locator(dicomViewer.webContentSelector).waitFor()
+    const stepUser = await getUser({user})
+    await dicomViewer.login({ username: stepUser.displayName, password: stepUser.password })
 })
 
-Given('{string} has uploaded the dicom file {string}', async function (user: string, filename: string): Promise<void> {
-    const page = state.page
+Given('the user has uploaded the dicom file {string}', async function (filename: string): Promise<void> {
     const dicomViewer = new DicomViewer()
+    await dicomViewer.upload({ filename })
+})
 
-    await dicomViewer.upload({filename: `${config.assets}/${filename}`})
-    await page.goto(config.baseUrlOcis)
-    await page.locator(util.format(dicomViewer.resourceNameSelector, filename)).waitFor()
+When('the user previews the dicom file {string}', async function (filename: string): Promise<void> {
+    const dicomViewer = new DicomViewer()
+    await dicomViewer.previewDicomFile({ filename })
 })
 
-When('{string} previews the dicom file {string}', async function (user: string, filename: string): Promise<void> {
+When('the user checks VIP metadata for dicom file', async function(): Promise<void> {
     const dicomViewer = new DicomViewer()
```

Comment:
is it necessary to create another instance of DicomViewer here? It doesn't look intuitive to me that the file is not passed to this instance but in the line below it is expected that the metadata that belongs to this file is displayed in the UI... maybe you can briefly chat to me to help me understand this :)

Reply:
Yes, we need to create a new instance in every step. Creating a new instance in every step will allow to interact with the UI element of the corresponding page. In this step, we have created an instance that will check if the element that contains the VIP metadata is visible. And, the following step creates an instance to check if the metadata, for instance, the patient's name, is correct or not.
Talking about whether the metadata belongs to the expected file, we create a new context in the `Before` hook as:
```
state.page = await context.newPage()
```
This will run before the test scenario and every step we implement like user log in, file upload, file previewing, etc. will happen within this context, which ensures that we are working on the same file.

Reply:
Thanks for explanations!